### PR TITLE
Make partitioned parquet reading deterministic

### LINKED ISSUE #4738: Non-deterministic partition parquet reading
I ran into a case where my partitioned parquet reading code would sometimes execute successfully, and sometimes error out.

The non-determinism comes in based on our logic which places special emphasis on the first path returned from `Files.newDirectoryStream`, which has javadoc:

> The elements returned by the iterator are in no specific order

At a minimum, we should be sorting the paths in some consistent fashion to ensure deterministic execution.

In my specific case, some of the parquet files were incorrectly written down w/ the partitioned columns, and when one of those files was the first, I would get the error:

```
RuntimeError: java.lang.IllegalArgumentException: Supplied ColumnDefinitions include duplicate names [KafkaPartition, Date]
	at io.deephaven.engine.table.TableDefinition.checkForNullOrDuplicates(TableDefinition.java:131)
	at io.deephaven.engine.table.TableDefinition.<init>(TableDefinition.java:118)
	at io.deephaven.engine.table.TableDefinition.of(TableDefinition.java:36)
	at io.deephaven.parquet.table.ParquetTools.readPartitionedTableInferSchema(ParquetTools.java:693)
	at io.deephaven.parquet.table.ParquetTools.readTableInternal(ParquetTools.java:585)
	at io.deephaven.parquet.table.ParquetTools.readTable(ParquetTools.java:97)
```

We could argue that this is "user error", but I suggest that we still want to be consistent and potentially improve how we handle sourcing the initial definition for partitioned parquet (maybe the user should be able to explicitly supply definition?).

### PR DESCRIPTION
**Pull request by devinrsmith**
This ensures that the firstEntryPath is the lexicographically first entry.

Part of #4738

### CODE REVIEW

**Comment by devinrsmith** [comment]
I'm actually doubting my change is complete looking back at it - it is only setting us up deterministically for ParquetKeyValuePartitionedLayout vs ParquetFlatPartitionedLayout, but _not_ how those work internally. I'm going to take another pass at this soon.

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java [hunk 1]
```diff
@@ -571,11 +572,13 @@ private static Table readTableInternal(
             }
             final Path firstEntryPath;
             try (final DirectoryStream<Path> sourceStream = Files.newDirectoryStream(sourcePath)) {
-                final Iterator<Path> entryIterator = sourceStream.iterator();
-                if (!entryIterator.hasNext()) {
+                // Lexicographical comparison
```
[the remark this answers is not in the document]
Comment (devinrsmith) [reply extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java:599]: So, the actual inference step is separate from this outer layer (readPartitionedTableInferSchema). My goal here was to make the existing behavior more deterministic; in this case, I think taking the min entry more closely matches the intentions and probable behavior of the existing code.

**Comment by devinrsmith** [comment]
I think there is still value in merging this PR to make things more deterministic, but it doesn't actually fix the inference step. Will leave that for later.

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcaudy) [reply extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java:599]: This doesn't make anything more deterministic, unless you read the DirectoryStream JavaDocs with no context. Directories have an order on every file system I'm aware of.

[the remark this answers is not in the document]
Comment (devinrsmith) [reply extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java:599]: This is _not_ true.

**Comment by malhotrashivam** [comment]
I think the change does what's it supposed to do. Do you think we should add a tiny test for this?

**Pull request closed without merging**